src/testsrc/java/com/epam/jamp2/service/TestCalculationCommandExecutionService.java

```diff
@@ -0,0 +1,133 @@
+package com.epam.jamp2.service;
+
+import com.epam.jamp2.model.CalculationCommand;
+import com.epam.jamp2.model.Operation;
+import com.epam.jamp2.model.UnknownCurrencyException;
+import com.epam.jamp2.model.Value;
+import com.epam.jamp2.service.impl.CalculationCommandExecutionServiceImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.stubbing.OngoingStubbing;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.hamcrest.beans.HasPropertyWithValue.hasProperty;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.number.BigDecimalCloseTo.closeTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(value = Parameterized.class)
+public class TestCalculationCommandExecutionService {
+
+    private CalculationCommandExecutionServiceImpl target;
+
+    private CalculationCommand command;
+    private Value expected;
+
+    private FxRatesService fxRatesService;
+
+    // Parameterized data for unit tests. Also used for ITCalculationCommandExecutionService.
+    @Parameters(name = "{index}: test calculate (command: {0}, {1}, {2}, {3}) should return = {4}")
```

Comment:
I think it would be better to have separate explicit tests for such cases as "should calc with same currencies", "should calc with different currencies" rather then covering them with parametrized data sets. Once such tests fails it will be absolutely clear to reader what happened while for parametrized test he will first need to analyze which scenario is represented by certain data set.